### PR TITLE
dark and light mode for chromium dialog

### PR DESCRIPTION
**Pull request by sujitacharya2005**
Resolves https://github.com/brave/brave-browser/issues/27767

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Dark Mode  | Light Mode
------------- | -------------
image  | image

### CODE REVIEW

File: android/java/res/drawable/notification_rationale_dialog_background.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) 2022 The Brave Authors. All rights reserved.
+     This Source Code Form is subject to the terms of the Mozilla Public
+     License, v. 2.0. If a copy of the MPL was not distributed with this file,
+     You can obtain one at http://mozilla.org/MPL/2.0/.
```
Comment (deeppandya) [inline android/java/res/drawable/notification_rationale_dialog_background.xml:5]: https

**Review by tapanmodh — APPROVED** [review]
lgtm

**Review by deeppandya — APPROVED** [review]
LGTM

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 6` running `Android 13` using the following build(s):

```
Brave | 1.49.41 Chromium: 109.0.5414.87 (Official Build) canary (32-bit)
--- | ---
Revision | 2dc18eb511c56e012081b4abc9e38c81c885f7d4-refs/branch-heads/5414@{#1241}
OS | Android 13; Build/TQ1A.230105.002
```

**`Light Theme`**

**`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|-----------------|---------------|----------------|----------------|-----------------|-----------------
Screenshot_20230118-220158 | Screenshot_20230118-220557 | Screenshot_20230118-220425 | Screenshot_20230118-220608 | Screenshot_20230118-220307 | Screenshot_20230118-220438 | Screenshot_20230118-220325 | Screenshot_20230118-223410 | Screenshot_20230118-223400

**`Dark Theme`**

**`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|-----------------|---------------|----------------|----------------|----------------|----------------
Screenshot_20230118-220756 | Screenshot_20230118-220841 | Screenshot_20230118-220856 | Screenshot_20230118-220949 | Screenshot_20230118-221057 | Screenshot_20230118-221111 | Screenshot_20230118-221223 | Screenshot_20230118-221235

**Pull request merged**